index.js

```diff
@@ -250,24 +250,35 @@ function refreshMyApps() {
 function getInstalledApps() {
   showLoadingIndicator();
   // Get apps
-  Comms.getInstalledApps().then(appIDs => {
+  return Comms.getInstalledApps().then(appIDs => {
     appsInstalled = appIDs;
     handleConnectionChange(true);
     refreshMyApps();
     refreshLibrary();
-  }).catch(err => {
-    showToast("Getting app list failed, "+err,"error");
   });
 }
 
 var connectMyDeviceBtn = document.getElementById("connectmydevice");
 
 function handleConnectionChange(connected) {
-  connectMyDeviceBtn.style.display = connected ? 'none' : '';
+  connectMyDeviceBtn.textContent = connected ? 'Disconnect' : 'Connect';
+  connectMyDeviceBtn.classList.toggle('is-connected', connected);
 }
 
-document.getElementById("myappsrefresh").addEventListener("click", getInstalledApps);
-connectMyDeviceBtn.addEventListener("click", getInstalledApps);
+document.getElementById("myappsrefresh").addEventListener("click", () => {
+  getInstalledApps().catch(err => {
+    showToast("Getting app list failed, "+err,"error");
+  });
+});
+connectMyDeviceBtn.addEventListener("click", () => {
+  if (connectMyDeviceBtn.classList.contains('is-connected')) {
```

Comment:
Here i have used a derived connection state and not the Puck.isConnected() state to avoid toggle/panic click race-conditions.
  
Would be better to manage this with a pending state but it would require state centralization and a bigger refactoring.